web/src/app/browser/src/languageMenu.ts

```diff
@@ -0,0 +1,582 @@
+   * @param   {boolean}   unique  is this the only keyboard for the language?
+   */
+  addKeyboard(kbd, kb, unique: boolean) {
+    kb.kn=kbd['KI'];        // InternalName;
+    kb.kc=kbd['KLC'];       // LanguageCode;
+    kb.innerHTML=unique?kbd['KL']:kbd['KN'].replace(' Keyboard',''); // Name
+
+    // We're setting up a few events - this alias helps avoid scoping issues.
+    const languageMenu = this;
+
+    // Refer to https://github.com/keymanapp/keyman/pull/7790 for context on
+    // the following two methods.
+    const lockBodyScroll = () => {
+      // If this object still exists, we never ran our paired `unlock` method;
+      // preserve the original state so that we can still restore it later!
+      if(this.originalBodyStyle) {
+        console.error("Unexpected state:  `originalBodyStyle` was not cleared by a previous `unlockBodyScroll()` call");
+        return;
+      }
+
+      // Preserve the original style for the body element; we're going to change
+      // it to block page scrolling.  Must use a separate instance.
+      //
+      // Reference: https://stackoverflow.com/a/28411556
+      this.originalBodyStyle = {};
+
+      // Must be separate line from previous due to TS type inference stuff.
+      const obs = this.originalBodyStyle;
+      const dbs = document.body.style;
+      obs.overflowY = dbs.overflowY;
+      obs.height = dbs.height;
+
+      // Now that the properties we're going to overwrite have been cached...
+      dbs.overflowY = 'hidden';
+      dbs.height = '100%';
+      return true;
+    }
+
+    const unlockBodyScroll = () => {
+      if(!this.originalBodyStyle) {
+        // We shouldn't be able to reach here, but in case things go out-of-order due
+        // to some unforeseen circumstance, let's null-guard here.
+        console.error("Unexpected state:  `originalBodyStyle` is unset; cannot restore original body style");
+        return;
+      }
+
+      // Reverses the changes to document.body.style made by `lockBodyScroll`.
+      const obs = this.originalBodyStyle;
+      const dbs = document.body.style;
+
+      dbs.overflowY = obs.overflowY;
+      dbs.height = obs.height;
+
+      // Successful restoration!  Clear the "restore to this" state so that the
+      // next 'lock' operation knows to do its part.
+      this.originalBodyStyle = null;
+    }
+
+    // Touchstart (or mspointerdown) event highlights the touched list item
+    const touchStart=function(e) {
+      e.stopPropagation();
+      if(this.className.indexOf('selected') <= 0) {
+        this.className=this.className+' selected';
+      }
+      languageMenu.scrolling=false;
+      languageMenu.y0=e.touches[0].pageY;//osk.lgList.childNodes[0].scrollTop;
+
+      lockBodyScroll();
+    };
+
+    //TODO: Still drags Android background sometimes (not consistently)
+    // Touchmove drags the list and prevents release from selecting the language
+    const touchMove=function(e: TouchEvent) {
```

Comment:
NOTE:  as in the original's implementation, this is _intentionally **not**_ an arrow function!  `this` will refer to the element the listener is set upon, and that's the intent of the original source.

Personally... I'd really rather rewrite that aspect.  But, maintaining it for now should help the review be more 'straightforward'.